frontend/src/lib/components/room/Chat.svelte

```diff
@@ -1,19 +1,24 @@
 <script lang="ts">
-  import { untrack } from 'svelte'
+  import { untrack, onDestroy } from 'svelte'
   import ky from '$lib/api/ky'
   import type { ChatMessage } from '$lib/api/room'
   import usernameCache from '$lib/state/usernameCache.svelte'
   import Textarea from '../Textarea.svelte'
+  import TypingIndicator from './TypingIndicator.svelte'
+  import type { SvelteMap } from 'svelte/reactivity'
 
   const systemUUID = '00000000-0000-0000-0000-000000000000'
 
   interface Props {
+    map: SvelteMap<string, number>
+    username: string
     disabled?: boolean
     messages: ChatMessage[]
     onSendMessage: (message: string) => void
+    onTyping: (map: SvelteMap<string, number>) => void
   }
-
-  const { messages, onSendMessage, disabled }: Props = $props()
+  let typingTimer: number | null = $state(null)
```

Comment:
We can remove the timer logic in this component after making the changes I suggest to `+page.svelte`